### PR TITLE
[master] Fix reception of ParameterList [6872]

### PR DESCRIPTION
**Pull request by IkerLuengo**
Port of #806

### CODE REVIEW

File: src/cpp/dds/core/policy/ParameterList.cpp [hunk 1]
```diff
@@ -179,39 +195,28 @@ bool ParameterList::readParameterListfromCDRMsg(CDRMessage_t& msg, std::function
...
-                    IF_VALID_CALL
+                    IF_VALID_CALL()
                 }
                 case PID_MULTICAST_IPADDRESS:
                 case PID_DEFAULT_UNICAST_IPADDRESS:
                 case PID_METATRAFFIC_UNICAST_IPADDRESS:
                 case PID_METATRAFFIC_MULTICAST_IPADDRESS:
                 {
-                    ParameterIP4Address_t p(pid, plength);
                     if (plength == PARAMETER_IP4_LENGTH)
```
Comment (MiguelCompany) [inline src/cpp/dds/core/policy/ParameterList.cpp]:
```suggestion
                    if (plength != PARAMETER_IP4_LENGTH)
```

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/942/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/783/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/939/)
* Performance Build Status

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/944/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/785/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/941/)
* Performance Build Status

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/955/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/794/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/952/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/279/)

**Comment by IkerLuengo** [comment]
As my understanding, all payloads in the message should be aligned to 32bit, adding padding if necessary. This was not being checked for TypeIdV1, and the tests were failing with the improved alignment processing on the readParameterListfromCDRMsg

**Pull request merged**